pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -41,23 +40,47 @@ func New(adapterContext common.AdapterContext, client kclient.Client) Adapter {
 		loggingClient = machineoutput.NewNoOpMachineEventLoggingClient()
 	}
 
+	componentInfoFactory := func(command versionsCommon.DevfileCommand) (common.ComponentInfo, error) {
+		pod, err := waitAndGetPod(true, adapterContext.ComponentName, client)
```

Comment:
I see the same thing called for supervisorFactory too. Can we move this pod variable and get it only once?

Reply:
I wasn't too sure, will look into it.

Reply:
Fixed.